### PR TITLE
Dev Container

### PR DESCRIPTION
**Pull request by esinx**
To make onboarding faster and trouble-free, we introduce dev containers. In short, onboarding newbies should be as easy as installing Docker *(or [OrbStack](orbstack.dev))*.

After installation, just launch dev container, run `pipenv install --dev` or `yarn` and everything will probably just work™️.

Even `docker compose` works thanks to docker-in-docker. It works surprisingly well with vscode's automatic port forwarding support.

Some caveats are:
- Make sure that your computer can handle multiple docker containers simultaneously
- Use `ssh-add` to "inherit" ssh credentials for source control
- Some initial setup (things like `./manage.py makemigrations`) remains to be done by the user
  - We can actually automate this, not sure if it's a good idea.

Screenshot 2024-02-29 at 4 53 47 PM

### CODE REVIEW

**Review by joyliu-q — APPROVED** [review]
Can you add documentation to the readme on Dev Container setup?
- Devcontainer set-up
- ssh add
- Other caveats mentioned

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM ubuntu:22.04
+ARG IMAGE_NAME=pennlabs/courses-devcontainer
+RUN apt-get update && apt-get install -y wget curl gcc python3-dev libpq-dev
```
Comment (joyliu-q) [inline .devcontainer/Dockerfile:3]: You should add git here

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+FROM ubuntu:22.04
```
Comment (judtinzhang) [inline .devcontainer/Dockerfile:1]: After testing some stuff on mobile, also suggest adding `--platform=linux/amd64` to be compatible with our prod machines running on x86 arch

Comment (esinx) [reply .devcontainer/Dockerfile:1]: I'm assuming this will fix dependency resolution? Sounds good to me!

**Pull request merged**